test/e2e/networking.go

```diff
@@ -71,140 +70,200 @@ var _ = Describe("Networking", func() {
 		}
 	})
 
-	//Now we can proceed with the test.
-	It("should function for intra-pod communication", func() {
-		if testContext.Provider == "vagrant" {
-			By("Skipping test which is broken for vagrant (See https://github.com/GoogleCloudPlatform/kubernetes/issues/3580)")
-			return
-		}
+	//Two parallel array - services[0] corresponds to timeouts[0]
+	services := [...]int{1, 2}         // the first test is equivalent to the original networking.go
+	timeouts := [...]float64{300, 600} // very liberal timeout policies.  Should be more like 100 seconds.
 
-		By(fmt.Sprintf("Creating a service named %q in namespace %q", svcname, f.Namespace.Name))
-		svc, err := f.Client.Services(f.Namespace.Name).Create(&api.Service{
-			ObjectMeta: api.ObjectMeta{
-				Name: svcname,
-				Labels: map[string]string{
-					"name": svcname,
-				},
-			},
-			Spec: api.ServiceSpec{
-				Ports: []api.ServicePort{{
-					Protocol:   "TCP",
-					Port:       8080,
-					TargetPort: util.NewIntOrStringFromInt(8080),
-				}},
-				Selector: map[string]string{
-					"name": svcname,
-				},
+	for i := range services {
+		ii := i
```

Comment:
Yes, good question.  The i gets overwritten in this loop, so using **i** leads to all tests running against the same (last) element of services.